python/src/scipp/compat/mantid.py

```diff
@@ -261,6 +288,30 @@ def convertEventWorkspace_to_dataarray(ws, load_pulse_times):
     return sc.DataArray(**coords_labs_data)
 
 
+def convertMDHistoWorkspace_to_dataset(md_histo):
+    ndims = md_histo.getNumDims()
+    if ndims > 3:
+        raise RuntimeError("Converter cannot process md histo workspace input "
+                           "with > 3 dimensions. Input has {}.".format(ndims))
```

Comment:
Why this limitation?

Reply:
Simply to limit the scope of the converter. I think given discussion here, I'll remove restriction.